com.unity.multiplayer.mlapi/Runtime/Transports/Transport.cs

```diff
@@ -148,7 +138,7 @@ protected void InvokeOnTransportEvent(NetEventType type, ulong clientId, byte ch
         /// <param name="payload">The incoming data payload</param>
         /// <param name="receiveTime">The time the event was received, as reported by Time.realtimeSinceStartup.</param>
         /// <returns>Returns the event type</returns>
-        public abstract NetEventType PollEvent(out ulong clientId, out byte channel, out ArraySegment<byte> payload, out float receiveTime);
```

Comment:
my biggest concern with the changes in this file is that we're touching it for naming/styling reasons but not getting actual performance or huge value out of it (not to disregard `byte` → `enum` but it's relatively minor compared to `string` → `byte` change we had previously which gave us quite a lot of memory footprint back, that was no-brainer to me).

we have quite a lot of other changes we want to make in this `Transport` abstraction but since it's also a quite sensitive area (apparently), we queued all those ideas and deferred until we have a plan.

also without `enum Channel` changes, files changed were around 4 IIRC but now 27 because it's touching a lot of places.

I'd rather hold this API change back for the time being and consider it in our `Transport` interface redesign/rework discussions later.

Reply:
Some great points.  Yet I like eliminating the manual built-in ids and making the channel a first-class type vs. byte by convention.  Also added some tests to make sure user-defined channels work (especially now that, uh, I understand how user-defined channels actually work :) )